framework/src/postprocessors/InterfaceDiffusiveFluxIntegral.C

```diff
@@ -0,0 +1,104 @@
+{
+  InputParameters params = InterfaceIntegralPostprocessor::validParams();
+
+  params.addRequiredCoupledVar("variable",
+                               "The name of the variable on the primary side of the interface");
+  params.addCoupledVar(
+      "neighbor_variable",
+      "The name of the variable on the secondary side of the interface. By default, "
+      "the primary side variable name is used for the secondary side");
+  params.addRequiredParam<MaterialPropertyName>(
+      "diffusivity", "The name of the diffusivity property on the primary side of the interface");
+  params.addParam<MaterialPropertyName>("neighbor_diffusivity",
+                                        "The name of the diffusivity property on the secondary "
+                                        "side of the interface. By default, the "
+                                        "primary side material property name is used for the "
+                                        "secondary side. Only needed for finite volume");
+  params.addClassDescription("Computes the diffusive flux on the interface.");
+
+  return params;
+}
+
+template <bool is_ad>
+InterfaceDiffusiveFluxIntegralTempl<is_ad>::InterfaceDiffusiveFluxIntegralTempl(
+    const InputParameters & parameters)
+  : InterfaceIntegralPostprocessor(parameters),
+    _grad_u(coupledGradient("variable")),
+    _u(coupledValue("variable")),
+    _u_neighbor(parameters.isParamSetByUser("neighbor_variable")
+                    ? coupledNeighborValue("neighbor_variable")
+                    : coupledNeighborValue("variable")),
+    _fv(getFieldVar("variable", 0)->isFV()),
+    _diffusion_coef(
+        getGenericMaterialProperty<Real, is_ad>(getParam<MaterialPropertyName>("diffusivity"))),
+    _diffusion_coef_neighbor(parameters.isParamSetByUser("neighbor_diffusivity")
+                                 ? getGenericNeighborMaterialProperty<Real, is_ad>(
+                                       getParam<MaterialPropertyName>("neighbor_diffusivity"))
+                                 : getGenericNeighborMaterialProperty<Real, is_ad>(
+                                       getParam<MaterialPropertyName>("diffusivity")))
+{
+
+  // Primary and secondary variable should both be of a similar variable type
+  if (parameters.isParamSetByUser("neighbor_variable"))
+    if ((_fv && !getFieldVar("neighbor_variable", 0)->isFV()) ||
+        (!_fv && getFieldVar("neighbor_variable", 0)->isFV()))
+      mooseError("For the InterfaceDiffusiveFluxIntegral, variable and "
+                 "neighbor_variable should be of a similar variable type.");
+
+  // Warn that we are not using the same gradient to compute the diffusive flux here
+  if (_fv && !isParamValid("neighbor_variable"))
+    mooseWarning("Only one finite volume variable was specified. InterfaceDiffusiveFluxIntegral is "
+                 "only accurate at a FVDiffusionInterface, not with a regular diffusion kernel.");
```

Comment:
I don't get why other conversation is outdated (I guess this filed moved?). But yea I think we should error here instead of potentially producing inaccurate results

Reply:
I m not sure if it moved. 
But there is no other option right now for computing the diffusive flux at an interface

Reply:
I guess I don't understand combination of the conditional and the message itself. This only warns if you're doing FV and the user didn't specify a `neighbor_variable`. So my response to the *conditional* is "well if it's wrong to do FV without specifying a `neighbor_variable` then we should just error and force them to specify a `neighbor_variable`". Is my thinking wrong? Does "not with a regular diffusion kernel" mean not with a `FVDiffusion` kernel? Because that on its own makes me think you're talking about FE or something (only the fact that we only get here when `_fv` makes me think you're talking about `FVDiffusion`).

Reply:
Scratch that last question ... I can see that your concern is almost assuredly about having different forms for the diffusive flux, e.g. `FVDiffusion` includes non-orthgonal contributions whereas `FVDiffusionInterface` only does orthogonal contributions (e.g. difference in centroid values)

Reply:
The user could be using the FVOneVarDiffusionInterface, or straight up FVDiffusion, and hoping to get the diffusive flux on an internal face. For the diffusive flux, we only have SideDiffusiveFlux and InterfaceDiffusiveFlux. The latter would be used in FE for the use cases mentioned, but in FV, we dont have an equivalent for them right now. 

It's not that inaccurate, it's still a centered approximation. I havent tested the order but I m going to claim second order right here right now ^^

Reply:
Umm I may be misunderstanding lemme check

Reply:
Ok the real problem is what kernels the user is using. This if statement is not catching all the bad options.

Reply:
ah actually they are not. Well I ll try using it on rod and see what it does

Reply:
Yea I agree that the core concern is about what kernel object they are using. This object will not exactly reproduce the flux computed by a non-orthogonal kernel. So yea I think trying to determine whether you've got a good pairing is certainly outside the scope of a postprocessor constructor. I would say that as long as you have documented in the `.md` and perhaps even in the class description (which will show up in the `.md` generally) that this object will only *exactly* reproduce the fluxes for an orthgonal-based diffusion approximation (and I guess specifically `FVDiffusionInterface`), then I think you should drop the warning and go on your way